code/modules/admin/admin_ranks.dm

```diff
@@ -100,165 +102,170 @@ GLOBAL_PROTECT(admin_ranks)
+		GLOB.admin_ranks += R
+		GLOB.protected_ranks += R
+		var/prev = findchar(line, "+-*", next, 0)
+		while(prev)
+			next = findchar(line, "+-*", prev + 1, 0)
+			R.process_keyword(copytext(line, prev, next), previous_rights)
+			prev = next
+		previous_rights = R.rights
+	if(!CONFIG_GET(flag/admin_legacy_system))
+		var/datum/DBQuery/query_load_admin_ranks = SSdbcore.NewQuery("SELECT rank, flags, exclude_flags, can_edit_flags FROM [format_table_name("admin_ranks")]")
 		if(!query_load_admin_ranks.Execute())
-			return
-		while(query_load_admin_ranks.NextRow())
-			var/rank_name = ckeyEx(query_load_admin_ranks.item[1])
-			var/flags = query_load_admin_ranks.item[2]
-			if(istext(flags))
-				flags = text2num(flags)
-			var/datum/admin_rank/R = new(rank_name, flags)
-			if(!R)
-				continue
-			GLOB.admin_ranks += R
-
+			message_admins("Error loading admin ranks from database. Loading from backup.")
+			log_sql("Error loading admin ranks from database. Loading from backup.")
+			CONFIG_SET(flag/admin_legacy_system, TRUE)
+			//load ranks from backup file
+			var/backup_file = file("data/admins_backup.json")
+			if(!fexists(backup_file))
+				log_world("Unable to locate admins backup file.")
+				return
+			var/list/json = json_decode(file2text(backup_file))
+			for(var/J in json["ranks"])
+				for(var/datum/admin_rank/R in GLOB.admin_ranks)
+					if(R.name == "[J]") //this rank was already loaded from txt override
+						continue
+				var/datum/admin_rank/R = new("[J]", json["ranks"]["[J]"]["include rights"], json["ranks"]["[J]"]["exclude rights"], json["ranks"]["[J]"]["can edit rights"])
+				if(!R)
+					continue
+				GLOB.admin_ranks += R
+			return 1
+		else
+			while(query_load_admin_ranks.NextRow())
+				var/skip
+				var/rank_name = query_load_admin_ranks.item[1]
+				for(var/datum/admin_rank/R in GLOB.admin_ranks)
+					if(R.name == rank_name) //this rank was already loaded from txt override
+						skip = 1
+						break
+				if(!skip)
+					var/rank_flags = text2num(query_load_admin_ranks.item[2])
+					var/rank_exclude_flags = text2num(query_load_admin_ranks.item[3])
+					var/rank_can_edit_flags = text2num(query_load_admin_ranks.item[4])
+					var/datum/admin_rank/R = new(rank_name, rank_flags, rank_exclude_flags, rank_can_edit_flags)
+					if(!R)
+						continue
+					GLOB.admin_ranks += R
 	#ifdef TESTING
 	var/msg = "Permission Sets Built:\n"
 	for(var/datum/admin_rank/R in GLOB.admin_ranks)
 		msg += "\t[R.name]"
-		var/rights = rights2text(R.rights,"\n\t\t",R.adds,R.subs)
+		var/rights = rights2text(R.rights,"\n\t\t")
 		if(rights)
 			msg += "\t\t[rights]\n"
 	testing(msg)
 	#endif
 
-
 /proc/load_admins()
+	var/dbfail
+	if(CONFIG_GET(flag/admin_legacy_system) && !SSdbcore.Connect())
+		message_admins("Failed to connect to database while loading admins. Loading from backup.")
+		log_sql("Failed to connect to database while loading admins. Loading from backup.")
+		CONFIG_SET(flag/admin_legacy_system, TRUE)
+		dbfail = 1

```

Comment:
I can see this being an annoyance if the issue is later fixed and there is no way to attempt to revert back to dbmins until the next round.